FixMultipleHerosWithSameTagBug

khaledmurtaja opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `alpha-x.x.x`: For stability testing: Only bug fixes accepted.
- `master`: Where the stable production ready code lies. Only security related bugs.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

`bugfix
`
**Issue Number:#1663**

**Did you add tests for your changes?**

I have tested the application manually after adding the `heroTag` attribute

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**
no, i have just added a missing attribute to a widget

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
This error was appearing in the debug console, everyone working on this project was wondering about the cause of this bug, on the technical side you mustn't have many **floating action buttons** with the same  hero tag, which will make the flutter not able to identify the FAB buttons, its like the `global keys` for widgets .
**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->
no breaking change.

literalEval commented:
@khaledmurtaja did you make sure that we are not referencing the tag you removed anywhere ? Also, with which widget was it clashing ?

khaledmurtaja commented:
> @khaledmurtaja did you make sure that we are not referencing the tag you removed anywhere ? Also, with which widget was it clashing ?

I did not remove any tags.
I have added one to the floating action button in the **Explore events screen**.
So why i have addeded one?
We have  two floating action buttons on  the Home screen.
**1**The first one is on the **Events tab**.
**2**The second one is on the **chats tab**.
Flutter needs to uniquely identify each tag so i have added a tag for one of them.
If we have added a new FAB in any new tab which belongs to the home screen we need to give it a **heroTag**.

literalEval commented:
@khaledmurtaja sorry I saw it on the flip side. Did you run the app and make sure that all the heroes are working fine without any warning ?

khaledmurtaja commented:
> @khaledmurtaja sorry I saw it on the flip side. Did you run the app and make sure that all the heroes are working fine without any warning ?

for sure, 
the error was appearing when i am trying to navigate to most of the screens inside the app. 
after the new commit, i was able to navigate without that error `multiple heros share the same tag`

literalEval commented:
Thanks for the info @khaledmurtaja
